addons/mail/models/mail_activity.py

```diff
@@ -181,31 +184,44 @@ def create(self, values):
         values_w_defaults = self.default_get(self._fields.keys())
         values_w_defaults.update(values)
 
+        # Reminder have no summary (for display name) It will add first line of note to summary.
+        if 'res_model_id' not in values:
+            text = html2plaintext(values.get('note', _('Reminder')))
+            values_w_defaults['summary'] = text.strip().replace('*', '').split("\n")[0]
+
         # continue as sudo because activities are somewhat protected
         activity = super(MailActivity, self.sudo()).create(values_w_defaults)
         activity_user = activity.sudo(self.env.user)
-        activity_user._check_access('create')
-        self.env[activity_user.res_model].browse(activity_user.res_id).message_subscribe(partner_ids=[activity_user.user_id.partner_id.id])
-        if activity.date_deadline <= fields.Date.today():
-            self.env['bus.bus'].sendone(
-                (self._cr.dbname, 'res.partner', activity.user_id.partner_id.id),
-                {'type': 'activity_updated', 'activity_created': True})
+        if activity.res_id and activity.res_model:
+            activity_user._check_access('create')
+            self.env[activity_user.res_model].browse(activity_user.res_id).message_subscribe(partner_ids=[activity_user.user_id.partner_id.id])
+            if activity.date_deadline <= fields.Date.today():
+                self.env['bus.bus'].sendone(
+                    (self._cr.dbname, 'res.partner', activity.user_id.partner_id.id),
+                    {'type': 'activity_updated', 'activity_created': True})
         return activity_user
 
     @api.multi
     def write(self, values):
         self._check_access('write')
         if values.get('user_id'):
             pre_responsibles = self.mapped('user_id.partner_id')
+
+        # Reminder have no summary (for display name) It will add first line of note to summary
+        if self._context.get('is_reminder') and 'note' in values:
```

Comment:
I really don't like having business related behavior using the context. Couldn't you deduce what to do depending on the activity status ? For example if I write on note on a reminder (aka no model, no res_id), update the summary if void. Do you think it would allow you to achieve the result without the use of context ?

Reply:
@tde-banana-odoo In case we get write on multiple records and if those records contains reminders+activity. In such case we need to call super write 2 times (though we don't have such case from UI)
 or 
we can simply stop such mix write from check_access 
can you give your feedback ?